domain/status/storage_test.go

```diff
@@ -69,6 +69,24 @@ func (s *storageStatusSuite) TestEncodeDecodeFilesystemStatus(c *tc.C) {
 	}
 }
 
+func (s *storageStatusSuite) TestFilesystemStatusTransitionErrorInvalid(c *tc.C) {
+	sts := StatusInfo[StorageFilesystemStatusType]{
+		Status: StorageFilesystemStatusTypeError,
+	}
+	err := FilesystemStatusTransitionValid(
+		StorageFilesystemStatusTypeAttached, true, sts)
+	c.Assert(err, tc.ErrorMatches, "cannot set status.*")
```

Comment:
does the error message form part of the contract that the caller expects? If not then we should be checking types and not strings.

This applies for all tests added to this file.

Reply:
Should have checked for error type